smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -109,12 +110,14 @@ abstract Schema createDocumentSchema(
 
     @Override
     public Optional<Operation> createOperation(Context<T> context, OperationShape operation) {
+        ServiceShape serviceShape = context.getService();
         return operation.getTrait(HttpTrait.class).map(httpTrait -> {
-            String method = context.getOpenApiProtocol().getOperationMethod(context, operation);
-            String uri = context.getOpenApiProtocol().getOperationUri(context, operation);
-            OperationObject.Builder builder = OperationObject.builder().operationId(operation.getId().getName());
             HttpBindingIndex bindingIndex = HttpBindingIndex.of(context.getModel());
             EventStreamIndex eventStreamIndex = EventStreamIndex.of(context.getModel());
+            String method = context.getOpenApiProtocol().getOperationMethod(context, operation);
+            String uri = context.getOpenApiProtocol().getOperationUri(context, operation);
+            OperationObject.Builder builder = OperationObject.builder()
+                    .operationId(serviceShape.getContextualName(operation));
```

Comment:
This shouldn't be needed since you can't rename operations, though there's not much reason to remove it now

Reply:
Yeah, this not needed, but I went ahead and used it in the places that it was possible just in case restrictions are ever lifted, and I think it serves to consistently show that you should use contextual names rather than shape names when generating "leaf-nodes" like OpenAPI or code.